From 3c8a69677b53562dcde0849f5497fd90e7bd358e Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@linux.ibm.com>
Date: Wed, 25 Apr 2018 17:29:26 +0200
Subject: [PATCH] crypt: correct s390 arch to include arch-specific crypto
 modules

Convert the s390x into s390 to also include s390-specific crypto
modules, for example, aes_s390 into the initramfs.

Signed-off-by: Hendrik Brueckner <brueckner@linux.ibm.com>
---
 modules.d/90crypt/module-setup.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/90crypt/module-setup.sh b/modules.d/90crypt/module-setup.sh
index b7232e98..1ec59096 100755
--- a/modules.d/90crypt/module-setup.sh
+++ b/modules.d/90crypt/module-setup.sh
@@ -27,6 +27,7 @@ installkernel() {
     hostonly="" instmods drbg
     arch=$(arch)
     [[ $arch == x86_64 ]] && arch=x86
+    [[ $arch == s390x ]] && arch=s390
     instmods dm_crypt =crypto =drivers/crypto =arch/$arch/crypto
 }
 

